src/Menu/Provider/GroupMenuProvider.php

```diff
@@ -134,10 +134,28 @@ private function canGenerateMenuItem(array $item, array $group)
         }
 
         //NEXT_MAJOR: Remove if statement of null checker.
-        return null === $this->checker || (
-            (empty($item['roles']) || $this->checker->isGranted($item['roles'])) &&
-            (empty($group['roles']) || $this->checker->isGranted($group['roles']))
-        );
+        if (null === $this->checker) {
+            return true;
+        }
+
+        // Making the checker behave affirmatively even if it's globally unanimous
+        $roles = null;
+        if (!empty($item['roles'])) {
+            $roles = $item['roles'];
+        } elseif (!empty($group['roles'])) {
```

Comment:
This means you can't have group + item roles defined...

Reply:
True, yet I'm following the same flow as before the change

Reply:
Not true, before your change, both item and group roles were checked.

Reply:
My bad, thanks for pointing this out. @jordisala1991 take a look again